### PR TITLE
update docs

### PR DESCRIPTION
**Pull request by DockToFuture**
**How to categorize this PR?**

/area networking
/kind enhancement

**What this PR does / why we need it**:
Documentation is updated to correctly describe now how the overlay is enabled and disabled.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```other operator
none
```

### CODE REVIEW

File: example/20-network.yaml [hunk 1]
```diff
@@ -46,11 +46,15 @@ spec:
...
-    ipam:
-      type: host-local
-      cidr: usePodCIDR
+    overlay:
+      enabled: false
+    snatToUpstreamDNS:
+      enabled: true
+#   ebpfDataplane:
+#     enabled: true
 #   ipAutodetectionMethod: interface=eth0
```
Comment (ScheererJ) [inline example/20-network.yaml:55]: Can you please also remove this entry? It seems deprecated and should not be included in the example anymore.

File: example/20-network.yaml [hunk 2]
```diff
@@ -46,11 +46,15 @@ spec:
...
-#   ipip: CrossSubnet
 #   typha:
 #     enabled: true
+#   ipam:
+#     type: host-local
+#     cidr: usePodCIDR
```
Comment (ScheererJ) [inline example/20-network.yaml:60]: Nit: Any particular reason why you moved `ipam` to the bottom?

**Review by ScheererJ — CHANGES_REQUESTED** [review]
Thanks. Looks good mostly.

File: example/20-network.yaml (hunk 2 above)
Comment (DockToFuture) [reply example/20-network.yaml:60]: I've never seen that anybody has set this field and do we really want stakeholder to set this?

Comment (ScheererJ) [reply example/20-network.yaml:59]: Ah, I did not get that you commented it. For sure, this is expert configuration, which should not necessarily be set. No issue to keep it commented out.

**Review by ScheererJ — APPROVED** [review]
/lgtm

**Pull request merged**